### PR TITLE
Add e2e tests

### PR DESCRIPTION
**Pull request by zjb0807**
Closes: #2648

### CODE REVIEW

File: ts-tests/tests/test-evm-fill-block.ts [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+
+        const creates = Array(15).fill(context.provider.api.tx.evm.create(
+            contract,
+            0,
+            2_000_000,
+            100_000,
+            []
+        ));
+
+        const tx = context.provider.api.tx.utility.batchAll(creates);
```
Comment (ermalkaleci) [inline ts-tests/tests/test-evm-fill-block.ts:34]: this is not the same as it was

Comment (zjb0807) [reply ts-tests/tests/test-evm-fill-block.ts:34]: `test-service` can manually generate blocks, send all of the transactions to mempool, and then generate blocks, but mandala cannot, so I use batchAll to test

Comment (ermalkaleci) [reply ts-tests/tests/test-evm-fill-block.ts:34]: can't we start manual-sealing, feed with tx and then build blocks and see how many tx are included per block?

File: ts-tests/tests/test-evm-call-fill-block.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        }
+
+        while (true) {
+            const currentHeight = await context.provider.api.query.system.number();
+
+            if (currentHeight.toNumber() > beforeHeight) {
+                break;
+            }
+
+            setTimeout(() => { }, 1000);
```
Comment (xlc) [inline ts-tests/tests/test-evm-call-fill-block.ts:42]: this doesn't wait, I think you want `await new Promise(resolve => setTimeout(resolve, 1000))`

File: ts-tests/tests/test-gas.ts [hunk 3]
```diff
@@ -18,7 +18,7 @@ describeWithAcala("Acala RPC (Gas)", (context) => {
 			from: alice.getAddress(),
 			data: "0x" + Block.bytecode,
 		});
-		expect(gas.toNumber()).to.be.eq(11301014);
+		expect(gas.toNumber()).to.be.eq(11301114);
```
Comment (xlc) [inline ts-tests/tests/test-gas.ts:21]: we should convert those into snapshot tests at some point

Comment (zjb0807) [reply ts-tests/tests/test-gas.ts:21]: use `closeTo` instead

**Review by xlc — APPROVED** [review]

**Review by xlc — APPROVED** [review]

**Pull request merged**